### PR TITLE
Add domain transfer to any user form for domain only sites

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/dotcom-forge/issues/3596

## Proposed Changes

* Add domain transfer to any user form for domain only sites

## Testing Instructions

Requires flag (applied by default in development env): `http://calypso.localhost:3000/domains/manage/all/test345678.blog/transfer/test345678.blog?flags=domains/transfer-to-any-user`

* Domain only site
* Click to transfer flow
* Should see new transfer to wordpress.com user option card
* Clicking through you should get the new transfer form
* Entering an email and clicking save should store the email in a blog option per D120926-code (requires patch applied)

Screenshot 2023-09-07 at 14-23-28 test345678 blog — WordPress com

### CODE REVIEW

File: client/my-sites/domains/domain-management/transfer/transfer-to-any-user/transfer-domain-to-any-user.tsx [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+		return (
+			<Card className="transfer-domain-to-any-user__card">
+				<form onSubmit={ handleSubmit }>
+					<FormFieldset>
+						<FormLabel>{ translate( 'Email of user to transfer to' ) }</FormLabel>
+						<FormTextInput
+							disabled={ isLoading }
+							name="email"
+							noWrap
```
Comment (candy02058912) [inline client/my-sites/domains/domain-management/transfer/transfer-to-any-user/transfer-domain-to-any-user.tsx]: I'm seeing this warning in the console
noWrap

**Review by candy02058912 — COMMENTED** [review]
Minor warning of React in the console, tested with sandbox, and it saved to the blog option.

File: client/my-sites/domains/domain-management/transfer/transfer-to-any-user/transfer-domain-to-any-user.tsx (hunk 1 above)
Comment (lsl) [reply client/my-sites/domains/domain-management/transfer/transfer-to-any-user/transfer-domain-to-any-user.tsx]: ~Not even sure why that's there!~ Copy pasta from domain forwarding 😄

```suggestion
```

**Review by candy02058912 — APPROVED** [review]
Warning not showing now 👍 LGTM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8925512

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8925512).

**Pull request merged**